packages/utils/src/calculate-gas-fees.ts

```diff
@@ -0,0 +1,50 @@
+import { BigNumber, constants, providers } from 'ethers';
+import { suggestFees } from 'eip1559-fee-suggestions-ethers';
+import Utils from '.';
+
+/**
+ * The function calculates gas fee information relative to EIP-1559.
+ * @param provider Generic provider.
+ * @param fallbackToGasPrice A boolean allowing if true to return gasPrice in case maxFeePerGas and maxPriorityFeePerGas equals 0.
+ * @param gasPriceMin Minimum gas price to return.
+ * @returns an object containing:
+ * - maxFeePerGas: The maximum fee per unit of gas for this transaction.
+ *   maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas
+ *   The baseFeePerGas depends on how full the previous blocks were.
+ * - maxPriorityFeePerGas: The maximum priority fee per unit of gas for this transaction.
+ * - gasPrice: Optional fallback: the gas price for this transaction.
+ */
+async function calculateGasFees(
+  provider: providers.JsonRpcProvider,
+  fallbackToGasPrice = false,
+  gasPriceMin?: BigNumber,
+): Promise<{
+  maxFeePerGas?: BigNumber;
+  maxPriorityFeePerGas?: BigNumber;
+  gasPrice?: BigNumber;
+}> {
+  const suggestedFee = await suggestFees(provider);
+
+  const baseFee = Utils.max(suggestedFee.baseFeeSuggestion, gasPriceMin || constants.Zero);
+
+  const maxPriorityFeePerGas = Utils.max(
+    suggestedFee.maxPriorityFeeSuggestions.urgent,
+    gasPriceMin || constants.Zero,
+  );
+  const maxFeePerGas = baseFee.add(maxPriorityFeePerGas);
+
+  if (maxPriorityFeePerGas.eq(0) || maxFeePerGas.eq(0)) {
+    if (fallbackToGasPrice) {
+      console.debug('Fallback to gasPrice');
+      return { gasPrice: await provider.getGasPrice() };
```

Comment:
Isn't it what is already done by the provider if we don't set any `gasPrice`?

Reply:
@olivier7delf I can confirm, this is not needed, see:
https://github.com/ethers-io/ethers.js/blob/c80fcddf50a9023486e9f9acb1848aba4c19f7b6/packages/abstract-signer/src.ts/index.ts#L272

Reply:
Indeed. 
I had done it before we want to handle gasPrice time to time, but it might make more sense to keep it outside the function...